src/DashboardService.Data/TaskRepository.cs

```diff
@@ -18,28 +21,104 @@ public TaskRepository(IDataProvider provider)
     _provider = provider;
   }
   
-  public Task<Guid?> CreateAsync(DbTask dbDepartment)
+  public async Task<Guid> CreateAsync(DbTask dbTask)
   {
-    throw new NotImplementedException();
+    _provider.Tasks.Add(dbTask);
+    await _provider.SaveAsync();
+
+    return dbTask.Id;
   }
 
-  public Task<List<DbTask>> GetAllAsync(GetTasksFilter filter)
+  public async Task<(List<DbTask> dbTasks, int totalCount)> GetAllAsync(GetTasksFilter filter, CancellationToken cancellationToken)
```

Comment:
add filtering by board, priority, type, deadline, creator. It will make it as flexible as possible

Reply:
++ de9aad33d8905e4f5a833bb5c1e0bef3767bd7e9